app/lib/common/providers/space_providers.dart

```diff
@@ -353,6 +359,64 @@ final remoteChatRelationsProvider =
   }
 });
 
+typedef RoomsAndRoomInfos = (List<String>, List<SpaceHierarchyRoomInfo>);
+
+final suggestedChatsProvider =
+    FutureProvider.family<RoomsAndRoomInfos, String>((ref, spaceId) async {
+  try {
+    final relatedSpaces =
+        await ref.watch(spaceRelationsOverviewProvider(spaceId).future);
+    final suggestedRooms = relatedSpaces.suggestedIds;
+    if (suggestedRooms.isEmpty) {
+      return (List<String>.empty(), List<SpaceHierarchyRoomInfo>.empty());
+    }
+    final toIgnore = relatedSpaces.knownChats.toList();
+    final localRooms = relatedSpaces.knownChats
+        .where((r) => suggestedRooms.contains(r))
+        .toList();
+    final roomHierarchy =
+        await ref.watch(spaceRemoteRelationsProvider(spaceId).future);
+    // filter out the known rooms
```

Comment:
To me it feels like this should all happen on the rust-side and if possible, the provider can yield unjoined or unknown chats/spaces which are relevant to the rooms we are part of. Something worth possible for future?